### DISCUSSION TITLE
PSRule.Rules.Azure/working-with-baselines/

### DISCUSSION THREAD

**Discussion by giscus** [Azure/PSRule.Rules.Azure#2539] [Documentation]
# PSRule.Rules.Azure/working-with-baselines/

This topic covers how to use the baselines shipped with PSRule for Azure.

https://azure.github.io/PSRule.Rules.Azure/working-with-baselines/

**Comment by kamfaima** [comment]
Is it mandatory to specify a baseline, and if not, what is the default assuming we're using the `PSRule.Rules.Azure` module?

Also, if the `ps-rule-assert@2` task can accept a comma separated string for modules, how does one load multiple baselines if, according to the [documentation](https://github.com/microsoft/PSRule-pipelines/blob/main/docs/tasks.md#ps-rule-assert) the `baselines:` input only accepts _"The name of a PSRule baseline to use."_ — the implication is it takes just a single baseline.

**Reply by BernieWhite** [reply]
@kamfaima Thanks for the question. It is optional to specify a baseline. But [working with baselines](https://azure.github.io/PSRule.Rules.Azure/working-with-baselines/) page gives you some reasons why you may would want to do that.

If you don't specify a baseline, then PSRule for Azure ships with a default baseline called [Azure.Default](https://azure.github.io/PSRule.Rules.Azure/en/baselines/Azure.Default/).

This baseline is updated automatically as new rules are released and includes all rules for generally available (GA) Azure features. This baseline does not include rules for Azure features or services that are under public/ private preview. This is also covered here https://azure.github.io/PSRule.Rules.Azure/working-with-baselines/#additional-standard-baselines.

New rules are added typically on a monthly basis at each new minor release.

---

Today it's not possible to run multiple baselines at the same time so `baseline:` parameter only supports a single baseline. We are considering this feature for PSRule v3 which is currently in development, however it may not be available at v3.0.0. If this is important to your use case we'd like to hear more about it here: https://github.com/microsoft/PSRule/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.md

---

Hope that helps.

**Reply by kamfaima** [reply]
Thanks for the prompt reply and apologies, I should have read the baseline docs a little more carefully!
